clients/src/test/java/org/apache/kafka/common/network/SSLSelectorTest.java

```diff
@@ -218,6 +222,49 @@ public void testRenegotiation() throws Exception {
         }
     }
 
+
+    /**
+     * Tests handling of BUFFER_UNDERFLOW during unwrap when network read buffer is smaller than SSL session packet buffer size.
+     */
+    @Test
+    public void testNetReadBufferResize() throws Exception {
+        String node = "0";
+        createSelector(10, 0, 0);
```

Comment:
Yes, that makes perfect sense. I was also considering suggesting that actually, but thought it was going to be more work for you. ;)

While you're at it, could you perhaps investigate if `SSLSelectorTest` could be more like `SslTopicMetadataTest` in that it doesn't duplicate the tests in `SelectorTest` that are the same (it inherits them from a shared base class in that case).